src/queue/BaseBatchedJob.php

```diff
@@ -106,18 +106,24 @@ final protected function totalItems(): int
      */
     final protected function totalBatches(): int
     {
-        return (int)ceil($this->totalItems() / $this->batchSize);
+        return (int)ceil($this->totalItems() / $this->batchSize());
+    }
+
+    final protected function batchSize(): int
+    {
+        return $this->batchSize ?? 1;
```

Comment:
This will effectively cause `batchSize` to default to `1`, meaning a new job will be spawned for every item. Which doesn’t seem consistent with the PR description.

My vote is to just revert the nullable `batchSize` change.

Reply:
Yeah, I guess our docs could just say to pass 1 if you want it to be controlled by ttr.

Reply:
Setting to `1` would have the same effect: a new batch per each individual item.

I don’t think there’s a realistic way to have an unlimited batch size that is fully left up to the TTR and memory consumption. We still need to know how many items to query at the beginning of `execute()` to iterate over for the current batch.

Reply:
oh duh, I was confusing batchSize and totalBatches.